stats/opentelemetry/client_metrics.go

```diff
@@ -232,8 +236,11 @@ func (h *clientStatsHandler) processRPCEnd(ctx context.Context, ai *attemptInfo,
 	}
 
 	for _, o := range h.options.MetricsOptions.OptionalLabels {
+		// TODO: Add a filter for converting to unknown if not present in the
+		// CSM Plugin Option layer by adding an optional labels API.
 		if val, ok := ai.xdsLabels[o]; ok {
 			attributes = append(attributes, otelattribute.String(o, val))
+			continue
```

Comment:
Why a continue when you're already at the end of the loop?

Reply:
Oh whoops, this is a remnant of when I had the default logic below. Deleted.